src/explorers/SliceSampler.jl

```diff
@@ -104,47 +131,52 @@ function slice_double(h::SliceSampler, state, z, pointer, log_potential, rng)
         K = K - 1
     end
     pointer[] = old_position # return the state back to where it was before
-    return(; L, R)
+    return (L, R, potent_L, potent_R)
 end
 
 function initialize_slice_endpoints(width, pointer, rng, ::Type{T}) where T <: AbstractFloat
     L = pointer[] - width * rand(rng)
     R = L + width
-    return(; L, R)
+    return (L, R)
 end
 
 function initialize_slice_endpoints(width, pointer, rng, ::Type{T}) where T <: Integer
     width = convert(T, ceil(width))
     L = pointer[] - rand(rng, 0:width)
     R = L + width 
-    return(; L, R)
+    return (L, R)
 end
 
 
 """
 $SIGNATURES
 Shrink the current slice.
 """
-function slice_shrink(h::SliceSampler, state, z, L, R, pointer, log_potential, rng)
+function slice_shrink!(h::SliceSampler, state, z, L, R, lp_L, lp_R, pointer, log_potential, rng)
     old_position = pointer[]
     Lbar = L
     Rbar = R
 
     while true
         new_position = draw_new_position(Lbar, Rbar, rng, typeof(pointer[]))
         pointer[] = new_position 
-        consider = z < log_potential(state)
+        new_lp = log_potential(state)
+        consider = z < new_lp
         pointer[] = old_position
-        if consider && slice_accept(h, state, new_position, z, L, R, pointer, log_potential)
-            return new_position
+        if consider && slice_accept(h, state, new_position, z, L, R, lp_L, lp_R, pointer, log_potential)
+            pointer[] = new_position
+            return new_lp
         end
         if new_position < pointer[]
             Lbar = new_position
         else
             Rbar = new_position
         end
     end
-    return new_position
+    # code should never get here...
```

Comment:
Good, point, and moreover, while looking at this code, I realized it is not immediately clear that this block of code is equivalent to the code in Neal's paper. Neal used a while loop of the form `while (rightProposalEndPoint - leftProposalEndPoint > 1.1 * initialWindowSize)` (the 1.1 is I think some number strictly between 1 and 2 to avoid round-off errors). Moreover I think the loop can exit and then the overall return is to accept. Probably equivalent to the code above but wanted to double check!

Reply:
Ah! just realized that's for the discrete sampler though..

Reply:
Then 

```
error()
return true
```

might be easier to understand?

Reply:
probably 
```
error()
return 0.0
```
to guarantee that the compiler thinks the return type is always float64

Reply:
oops yes sorry!